File: doc/porting/MetalCrateItem.cs
using Assets.Scripts.GamePlayScene;
using Assets.Scripts.GamePlayScene.Mechanics;
using Assets.Scripts.GamePlayScene.Mechanics.Items;
using Assets.Scripts.Utils;
using System;

namespace GamePlayScene.Mechanics.Items
{
	public class MetalCrateItem : SpriteBasedItem
	{
		public override ItemType GetItemType()
		{
			return ItemType.MetalCrate;
		}

		public override int GetScore()
		{
			return 1000;
		}

		public override bool CanCastShadow()
		{
			return false;
		}

		public override bool CanShadowDrop()
		{
			return true;
		}

		public override bool CanFall()
		{
			return false;
		}

		public override void PlayExplodeAudio()
		{
			AudioManager.Play(AudioTag.MetalCrate, PlayMode.Frame, null, 1f);
		}

		protected override void PlayOnExplodeStartedAnimation()
		{
			if (ParticlePool.Instance != null)
			{
				ParticlePool.Instance.PlayParticle(ParticlePool.Instance.MetalCrateParticle, base.get_transform().get_position(), true);
			}
		}
	}
}

File: classes/game/item/MetalCrateItem.cpp
#include "MetalCrateItem.h"
#include "../Animation.h"
#include "../ParticleLibrary.h"
#include "../ImageLibrary.h"
#include "global/SoundManager.h"

void MetalCrateItem::SetupItem() {
	ImageLibrary::CopySprite(this->CurrentSpriteRenderer, IMGLIB->MetalCrate);
}

void MetalCrateItem::RemoveSelf() {
	this->OnBeforeRemove();
	//base.get_gameObject().Recycle();
	this->CurrentSpriteRenderer->removeFromParentAndCleanup(true);
	//delete this;
}

void MetalCrateItem::PlayExplodeAudio() {
	//AudioManager.Play(AudioTag.MetalCrate, PlayMode.Frame, null, 1f);
}


void MetalCrateItem::PlayOnExplodeStartedAnimation() {
	//if(ParticlePool.Instance != null) {
	//	ParticlePool.Instance.PlayParticle(ParticlePool.Instance.MetalCrateParticle, base.get_transform().get_position(), true);
	//}
	PTCLLIB->explodeMetalCrate(CurrentSpriteRenderer->getPosition());
}